be/src/http/action/stream_load_with_sql.cpp

```diff
@@ -0,0 +1,548 @@
+                memcpy(bb->ptr, ctx->schema_buffer + remove_bytes, cur_remove_bytes);
+                ctx->schema_buffer_size -= cur_remove_bytes;
+                remove_bytes += cur_remove_bytes;
+                bb->pos = cur_remove_bytes;
+                ctx->receive_bytes += cur_remove_bytes;
+                bb->flip();
+                auto st = ctx->body_sink->append(bb);
+                if (!st.ok()) {
+                    LOG(WARNING) << "append body content failed. errmsg=" << st << ", "
+                                 << ctx->brief();
+                    ctx->status = st;
+                    return;
+                }
+            }
+            ctx->restore_pipe_promise.set_value(ctx->status);
+            ctx->need_schema_buffer = false;
+            ctx->need_wait_restore_pipe = false;
+        }
+        ctx->read_data_cost_nanos += (MonotonicNanos() - start_read_data_time);
+    } else {
+        // local file no need to buffer
+        while (evbuffer_get_length(evbuf) > 0) {
+            auto bb = ByteBuffer::allocate(stream_buffer_size);
+            auto remove_bytes = evbuffer_remove(evbuf, bb->ptr, bb->capacity);
+            bb->pos = remove_bytes;
+            bb->flip();
+            auto st = ctx->body_sink->append(bb);
+            if (!st.ok()) {
+                LOG(WARNING) << "append body content failed. errmsg=" << st << ", " << ctx->brief();
+                ctx->status = st;
+                return;
+            }
+            ctx->receive_bytes += remove_bytes;
+        }
+        ctx->read_data_cost_nanos += (MonotonicNanos() - start_read_data_time);
+    }
+}
+
+void StreamLoadWithSqlAction::free_handler_ctx(std::shared_ptr<void> param) {
+    std::shared_ptr<StreamLoadContext> ctx = std::static_pointer_cast<StreamLoadContext>(param);
+    if (ctx == nullptr) {
+        return;
+    }
+    // sender is gone, make receiver know it
+    if (ctx->body_sink != nullptr) {
+        ctx->body_sink->cancel("sender is gone");
+    }
+    // remove stream load context from stream load manager and the resource will be released
+    ctx->exec_env()->new_load_stream_mgr()->remove(ctx->id);
+}
+
+Status StreamLoadWithSqlAction::_process_put(HttpRequest* http_req,
+                                             std::shared_ptr<StreamLoadContext> ctx) {
+    // put request
+    TStreamLoadPutRequest request;
+    set_request_auth(&request, ctx->auth);
+    request.txnId = ctx->txn_id;
+    request.__set_version(1);
+    request.__set_load_sql(http_req->header(HTTP_SQL));
+    request.__set_loadId(ctx->id.to_thrift());
+    request.__set_label(ctx->label);
+    if (_exec_env->master_info()->__isset.backend_id) {
+        request.__set_backend_id(_exec_env->master_info()->backend_id);
+    } else {
+        LOG(WARNING) << "_exec_env->master_info not set backend_id";
+    }
+    if (!http_req->header(HTTP_EXEC_MEM_LIMIT).empty()) {
```

Comment:
It is useless. SQL based load should use  set_var session variable to config exec memlimit;